### ISSUE TITLE
Packer VirtualBox builds failing under macOS High Sierra 10.13.2

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/packer-ubuntu-1604#35]
So, super strange thing: I can't build this box—or any of my other Debian/Ubuntu boxes, for that matter—on my work laptop, running macOS 10.13.2. So far I've only debugged on that laptop, running both VirtualBox 5.2.4 and VirtualBox 5.1.30, along with both Packer 1.0.3, and 1.1.3.

For some reason, Packer itself is really slow no matter the version, so I'm wondering if there's something on the system hindering the build.

The `boot_command` was highly suspect (Ubuntu's always done weird things with it anyways...), because for some reason in one case the installer would drop to a blank screen (nothing on it), and then eventually it would drop back to language selection.

In other cases (playing with the `<esc><wait><f6><wait>` etc game in the beginning of the `boot_command`), I would get a popup for 'exit to the console installer' over and over again. Very odd behavior.

So using `PACKER_LOG=1 packer build --only=virtualbox-iso ubuntu1604.json`, I found that just using the following `boot_command`:

```json
  "builders": [
    {
      "type": "virtualbox-iso",
      "boot_command": [
        "<enter>"
      ],
      ...
```

I would get the following output in the console on my host:

```
2017/12/24 22:59:49 ui: ==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Typing the boot command...
2017/12/24 22:59:49 packer: 2017/12/24 22:59:49 Special code '<enter>' found, replacing with: [1c 9c]
2017/12/24 22:59:49 packer: 2017/12/24 22:59:49 Executing VBoxManage: []string{"controlvm", "packer-ubuntu-16.04-amd64", "keyboardputscancode", "1c"}
2017/12/24 22:59:52 packer: 2017/12/24 22:59:52 stdout:
2017/12/24 22:59:52 packer: 2017/12/24 22:59:52 stderr:
2017/12/24 22:59:52 packer: 2017/12/24 22:59:52 Executing VBoxManage: []string{"controlvm", "packer-ubuntu-16.04-amd64", "keyboardputscancode", "9c"}
2017/12/24 22:59:55 packer: 2017/12/24 22:59:55 stdout:
2017/12/24 22:59:55 packer: 2017/12/24 22:59:55 stderr:
2017/12/24 22:59:55 packer: 2017/12/24 22:59:55 [INFO] Waiting for SSH, up to timeout: 2h46m40s
2017/12/24 22:59:55 ui: ==> virtualbox-iso: Waiting for SSH to become available...
==> virtualbox-iso: Waiting for SSH to become available...
```

...but when switching off headless mode (so I could see the GUI), here's what I saw:

![packer-ubuntu-weirdness](https://user-images.githubusercontent.com/481677/34333204-a7dec938-e8ff-11e7-98a5-bd4cbf4cf427.gif)

**Comment by geerlingguy** [comment]
So super strange thing, too...

`1c 9c` is the keydown + keyup for the enter key... and if I run the following commands myself after going back to the installer main menu in the running vm:

```
VBoxManage controlvm packer-ubuntu-16.04-amd64 keyboardputscancode 1c
VBoxManage controlvm packer-ubuntu-16.04-amd64 keyboardputscancode 9c
```

Then it proceeds with installation and gets ready for the preseed file... so something's going funny with the interpretation of the keycodes, but only when they come from Packer. Maybe my terminal environment is jacking up the text somehow??

**Comment by geerlingguy** [comment]
I'm dropping my own .bash_profile customizations in a new shell, and will see if this makes any difference. Could there be some other text encoding going on between Packer and the shell?

**Comment by geerlingguy** [comment]
Don't know if this is related in any way, but https://github.com/hashicorp/packer/issues/2855.

I'm switching way back to Packer 0.12.2, which I know has worked on this particular computer in the past, and will see if there's any positive difference. Next test will be running the exact same config on my personal laptop (same config, same OS, different hardware).

And if that doesn't work, I'll try it on Fedora. If that doesn't work, Windows... gotta find somewhere this works.

**Comment by geerlingguy** [comment]
Also, right when the commands are passed, while watching Apple's Console app, I see the message:

```
VirtualBoxVM (LaunchServices)
LSExceptions shared instance invalidated for timeout.
```

And a second before that some messages like:

```
IOHIDDeviceClass::start: elementCount=16 reportHandlerCount=4
IOHIDDeviceClass::buildElements: type=0 *buffer=7f91cb881030 *count=16 size=1536
IOHIDDeviceClass::buildElements: type=1 *buffer=7f91cc4334c0 *count=4 size=384
IOHIDDeviceClass::start: elementCount=16 reportHandlerCount=4
IOHIDDeviceClass::buildElements: type=0 *buffer=7f91cb882a30 *count=16 size=1536
IOHIDDeviceClass::buildElements: type=1 *buffer=7f91cc42d410 *count=4 size=384
```

And interestingly, a few of these:

```
Month 13 is out of bounds
```

**Comment by geerlingguy** [comment]
Trying Packer 1.1.3 with Virtualbox latest dev build 5.2.97 (119762 - from https://www.virtualbox.org/wiki/Testbuilds)

**Comment by geerlingguy** [comment]
Hmm, found this in the VirtualBox log (in the VM folder, in my case `~/VirtualBox VMs/[vm name]/Logs/VBox.log`):

```
00:00:11.806103 VMMDev: Guest Log: BIOS: KBD: unsupported int 16h function 03
00:00:11.806513 VMMDev: Guest Log: BIOS: AX=0305 BX=0000 CX=0000 DX=0000
```

Found https://www.virtualbox.org/ticket/8058, but it doesn't seem necessarily related... checking all the created VM settings seems to indicate things are set up correctly.

**Comment by geerlingguy** [comment]
Strange, if I just hit 1c... but not 9c, then the install starts going—but I think that's because it keeps hitting enter on the 'install ubuntu' screen then every dialog that comes afterwards.

Okay, changing the game a bit and will try on another computer.

**Comment by geerlingguy** [comment]
Welp.

Ran it on my personal laptop. Same exact versions of everything... and no issues whatsoever.

I wonder if the work laptop has some sort of networking or other kind of proxying system installed that's screwing with commands and VirtualBox? Maybe some cache somewhere?

Well, since I can't reproduce this issue on another computer, I'm probably going to wipe my work laptop and reinstall. Probably faster to do that (~3-4 hours, all [automated](https://github.com/geerlingguy/mac-dev-playbook)) than debug exactly where this error is coming from.

**Comment by aspyatkin** [comment]
Hello, did you manage to resolve this problem?
I am facing the same problem - can't build Ubuntu 16.04 images with recent versions of Packer (1.1.3), Virtualbox (5.2.4), macOS 10.13.2. For some reason it just quits console mode installation and shows the same "Select language" window as in the screenshot above. Tried different boot commands from several packer template repositories, with no luck at all. I clearly remember I was able to build an image using Virtualbox 5.1.24 and Packer 1.0.3 ([template](https://github.com/aspyatkin/ubuntu-16.04-packer/blob/master/template.json)).
